replication.go

```diff
@@ -24,12 +24,24 @@ package driver
 
 import (
 	"context"
+	"time"
 )
 
+type Tick string
+
+type BatchMetadata struct {
+	ID       string `json:"id"`
+	LastTick Tick `json:"lastTick,omitempty"`
+}
+
 // Replication provides access to replication related operations.
 type Replication interface {
+	// CreateBatch creates a "batch" to prevent WAL file removal and to take a snapshot
+	CreateBatch(ctx context.Context, serverID int64, db Database, ttl time.Duration) (BatchMetadata, error)
+	// DeleteBatch deletes an existing dump batch
+	DeleteBatch(ctx context.Context, db Database, batchID string) error
 	// Get the inventory of the server containing all collections (with entire details) of a database.
 	// When this function is called on a coordinator is a cluster, an ID of a DBServer must be provided
 	// using a context that is prepare with `WithDBServerID`.
-	DatabaseInventory(ctx context.Context, db Database) (DatabaseInventory, error)
+	DatabaseInventory(ctx context.Context, db Database, batchID string) (DatabaseInventory, error)
```

Comment:
This is a breaking change in the API!
If a batchID has to be brought in (it is certainly not needed in all cases), use a context variable (e.g. `WithBatchID`)
See context.go

Reply:
The batchID is needed to get the inventory when we use rocksdb. So this API could not have worked for anybody (using rocksdb), I do not see it as a huge issue to add a new method.

Maybe that line of thought is wrong ?

Reply:
In Arangosync the DatabaseInventory is already used. It calls a dbserver directly.

Reply:
What do you mean add a new method?

Reply:
Sorry I meant write "add a new argument". Arangosync uses `DatabaseInventory` from the `Cluster` interface, no?

Reply:
Again, adding a new argument is breaking is API. Not good.
Please use the context way.